File: habit_backend/models/customizationItem.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");
const User = require("./user");

const CustomizationItem = sequelize.define("customization_item", {
  name: { type: DataTypes.STRING, allowNull: false },
  type: { type: DataTypes.STRING, allowNull: false },
  level_required: { type: DataTypes.INTEGER, defaultValue: 1 },
  points_cost: { type: DataTypes.INTEGER, defaultValue: 0 },
});

CustomizationItem.belongsToMany(User, { through: "user_customization_items" });
User.belongsToMany(CustomizationItem, { through: "user_customization_items" });

module.exports = CustomizationItem;

File: front-flutter/lib/models/customization_item.dart
class CustomizationItem {
  final int id;
  final String name;
  final String type;
  final int levelRequired;
  final int pointsCost;

  CustomizationItem({
    required this.id,
    required this.name,
    required this.type,
    required this.levelRequired,
    required this.pointsCost,
  });

  factory CustomizationItem.fromJson(Map<String, dynamic> json) {
    return CustomizationItem(
      id: json['id'],
      name: json['name'],
      type: json['type'],
      levelRequired: json['level_required'],
      pointsCost: json['points_cost'],
    );
  }
}
